src/components/include/DefaultImportModalDialog.jsx

```diff
@@ -0,0 +1,323 @@
+import nativeFormats from "@exabyte-io/made.js/lib/parsers/native_formats";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import FormControl from "@mui/material/FormControl";
+import { DataGrid } from "@mui/x-data-grid";
+import PropTypes from "prop-types";
+import React from "react";
+import NPMsAlert from "react-s-alert";
+import _ from "underscore";
+
+import { Material } from "../../material";
+
+class DefaultImportModalDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            fileNames: [],
+            formats: [],
+            texts: [],
+            dragging: false,
+        };
```

Comment:
let's use `useState` hook for each state field separate setter

Reply:
Agree on this as well.